cmake/dts.cmake

```diff
@@ -28,6 +66,7 @@ if(CONFIG_HAS_DTS)
   set(
     dts_files
     ${DTS_SOURCE}
+    ${DTC_SHIELD_OVERLAY_FILES_AS_LIST}
```

Comment:
We can't modify core build scripts every time there is a new use-case. We must strive to re-use a few existing concepts instead of adding more and more complexity.

Could you look into if it is possible, to get the shield overlay files into DTC_OVERLAY_FILE_AS_LIST somehow to avoid this (and similair) patches?

And if not possible, elaborate on why not it cannot be done so we better understand the limitations of the DTS infrastructure. (And so that reviewers can easily verify that we have done all that is possible to minimize complexity).

Reply:
Agreed. I added this new variable initially for a better control of my change, and be sure I don't mess with DTC_OVERLAY_FILE variable precedence. But now I should be able to do it w/o DTC_OVERLAY_FILE_AS_LIST